registry/handlers/app.go

```diff
@@ -678,6 +671,21 @@
 
 		context := app.context(w, r)
 
+		defer func() {
+			// Automated error response handling here. Handlers may return their
+			// own errors if they need different behavior (such as range errors
+			// for layer upload).
+			if context.Errors.Len() > 0 {
+				if err := errcode.ServeJSON(w, context.Errors); err != nil {
+					dcontext.GetLogger(context).Errorf("error serving error json: %v (from %v)", err, context.Errors)
```

Comment:
This seems completely bogus.

Reply:
CodeQL seems to suggest that logging of this sensitive info might happen in registry proxy [and swift driver] code. I need to dig in further properly

Reply:
Thanks in advance for taking a look at that. Besides the false positive from CodeQL, does the PR look good?

Reply:
Other than that the PR looks good yeah. I'd like to understand these CodeQL warnings better though. I have not had any time to track the code or test manually what happens when an error is triggered in the proxy mode. Equally if this is a FP I wonder if there is a way to tell CodeQL to stfu.

<img width="778" alt="Screenshot 2022-09-28 at 09 26 39" src="https://user-images.githubusercontent.com/1392526/192729064-33c65a75-6022-483a-92d8-b28e0dba3dc5.png">

Reply:
Just a gentle reminder about this; would be great to get this PR past the review process so I can unload the mental context 😃

Reply:
@milosgajdos: ping?

Reply:
we need to find a way how to shush codeQL scanner -- otherwise the builds will keep on failing from the moment we merge this; also I'm not entirely sure if we can merge anything with codeQL scanner barfing (I think there is some weird CNCF policy or something)

Reply:
@milosgajdos: Any chance we can work around the CodeQL scanner and get this merged? I can just remove this log statement if that's the best path forward.

Reply:
Yeah I was thinking a while back about whether removing this log statement could be the best course of action 🤔

Reply:
Done.